src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_CAPACITY = 10;
+    private Object[] array = new Object[MAX_CAPACITY];
```

Comment:
1. Better to parametrize array in the beginning to avoid type-casts in methods
2. It's not max capacity, our array can be much longer after resizes
```suggestion
    private T[] array = (T) new Object[DEFAULT_CAPACITY];
```